### PR TITLE
[ptf]: Add ptf service in SONiC neighbor devices

### PR DESCRIPTION
**Pull request by Pterosaur**
Signed-off-by: Ze Gan <ganze718@gmail.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Add ptf service in SONiC neighbor devices to enhance the power of testbed so that we have an opportunity to proactively send packet from SONiC neighbor devices or check the receiving packet on the devices.

#### How did you do it?
1. Save the ptf image from host machine in add_topo stage
2. Dispatch ptf image to the neighbor devices in add_topo stage
3. Start ptf services in the neighbor devices and register them in the ptfadapter in test stage

### CODE REVIEW

File: ansible/roles/vm_set/tasks/add_topo.yml [hunk 1]
```diff
@@ -225,3 +225,10 @@
     when: "'dualtor' in topo"
 
   when: container_type == "PTF"
+
+- name: Save PTF image
+  block:
+  - shell: docker tag "{{ docker_registry_host }}/{{ ptf_imagename }}:{{ ptf_imagetag }}" docker-ptf
+  - shell: docker save -o docker-ptf.tar docker-ptf
+  - fetch:  src=docker-ptf.tar dest=docker-ptf.tar flat=yes
+  run_once: yes
```
Comment (wangxin) [inline ansible/roles/vm_set/tasks/add_topo.yml:234]: What if there is already a file with name 'docker-ptf.tar' on disk? Will it overwrite or raise error?

File: ansible/roles/sonic/tasks/vsonic.yml [hunk 2]
```diff
@@ -64,3 +64,8 @@
   when: hostname in configuration
   notify:
     - Restart BGP service
+
+- name: Load PTF image
+  block:
+    - copy: src="docker-ptf.tar" dest="docker-ptf.tar"
+    - shell: docker load -i docker-ptf.tar || true
```
Comment (wangxin) [inline ansible/roles/sonic/tasks/vsonic.yml:71]: Cleanup the docker-ptf.tar file from local and remote after "docker load -i" is done?

File: ansible/roles/vm_set/tasks/add_topo.yml (hunk 1 above)
Comment (Pterosaur) [reply ansible/roles/vm_set/tasks/add_topo.yml:234]: It will overwrite it

File: ansible/roles/sonic/tasks/vsonic.yml (hunk 2 above)
Comment (Pterosaur) [reply ansible/roles/sonic/tasks/vsonic.yml:71]: Done

**Review by wangxin — APPROVED** [review]

**Pull request merged**